bool success
string msg
int64 path_id
